fedimint-client/src/sm/executor.rs

```diff
@@ -656,11 +698,22 @@ impl ExecutorInner {
                                                     .get(&state.module_instance_id())
                                                     .expect("Unknown module");
 
+                                                let operation_id = state.operation_id();
                                                 let global_context = global_context_gen(
                                                     state.module_instance_id(),
-                                                    state.operation_id(),
+                                                    operation_id,
                                                 );
-                                                if new_state.is_terminal(context, &global_context) {
+
+                                                let is_terminal = new_state.is_terminal(context, &global_context);
+                                                    self.log_event_dbtx(dbtx,
+                                                        StateMachineUpdated{
+                                                            operation_id,
+                                                            module_id: state_module_instance_id,
+                                                            terminal: is_terminal,
+                                                            outcome: transition_outcome
+                                                        }
+                                                    ).await;
+                                                if is_terminal {
```

Comment:
What's going on with the indentation here?

Reply:
rust-analyzer sometimes can't, but the part you highlighted looks OK to me?